src/main/java/nl/inl/corpuswebsite/utils/XslTransformer.java

```diff
@@ -29,16 +31,25 @@
 
     private final Transformer transformer;
 
+    private static final Map<File, Templates> FILETEMPLATES = new ConcurrentHashMap<File, Templates>() {
+        @Override
+        public Templates get(Object key) {
+            if (super.get(key)==null) {
+                try {
+                    put((File)key, FACTORY.newTemplates(new StreamSource((File) key)));
+                } catch (TransformerConfigurationException ex) {
+                    throw new RuntimeException(ex);
+                }
+            }
+            return super.get(key);
+        }
+        
```

Comment:
I don't know about overriding a Map implementation like this. At the very least you should also override containsKey(), entrySet(), etc. to avoid breaking the Map contract, but I think this is a classic case where you should prefer encapsulation over inheritance.